scripts/collect_api_events.rb

```diff
@@ -38,38 +44,55 @@ def parse_body(response_body)
   body
 end
 
-config_file = ENV['CONFIG_FILE'] || "#{File.expand_path(File.dirname(__FILE__))}/../conf/splunk_config.yaml"
-settings = load_settings(config_file)
-
-splunk_client = CommonEventsHttp.new('http://' + settings['splunk']['server'], port: settings['splunk']['port'], ssl_verify: false)
-orchestrator = Orchestrator.new(settings['pe']['console'], settings['pe']['username'], settings['pe']['password'], ssl_verify: false)
-response = orchestrator.get_all_jobs
-raise "Failed to get the jobs from PE [#{response.error!}]" unless response.code == '200'
+def store_index(index, filepath)
+  open(filepath, 'w') { |f| f.puts index }
+end
 
-body = parse_body(response.body)
+def get_index(filepath)
+  return 0 unless File.file?(filepath)
+  File.readlines(filepath).each do |line|
+    return line.to_i
+  end
+end
 
-events_json = transform(body['items'], settings['pe']['console'], 'puppet:summary')
+def process_response(body, total, settings, source_type, splunk_client)
+  return false if total.zero? || total.nil?
 
-if events_json.empty?
-  puts 'There were no orchestrator events'
-else
+  events_json = transform(body, settings['pe']['console'], source_type, splunk_client, settings['splunk']['token'])
+  store_index(body.empty? ? 0 : total, API_EVENTS_STORE)
```

Comment:
Is using the `API_EVENTS_STORE` constant going to store the index in the same store file no matter what? Don't we need to use a different file for events vs activities?